### PR TITLE
Handle "@submit" events on forms in dialog boxes

### PR DESCRIPTION
**Pull request by joepavitt**
Closes https://github.com/flowforge/flowforge/issues/334

### CODE REVIEW

**Comment by knolleary** [comment]
The original problem is that hitting enter in an `<input>` is causing the form to be submitted - which reloads the page rather than save the changes as expected. That sucked and needed fixing. However I'm not 100% on whether this approach is the right one.

My concern is with the approach of this fix is that now hitting enter in *any* input is now the same as clicking the submit button. If I have a form with 5 inputs I may hit enter in the first input expecting to move to the next input and carry on filling in the form, whereas this will trigger the completion of the form. That can lead to premature form submission - and given we have some forms with fields that cannot be edited after they are set, that could lead to unexpected/frustrating behaviour.

Would it be better to just default the submit action all together, and *only* complete the form when the appropriate button is clicked (which can be achieved via tab navigation and hitting 'enter' to ensure keyboard accessibility).

**Comment by joepavitt** [comment]
> Would it be better to just disable the submit action all together, and only complete the form when the appropriate button is clicked

I am okay with this

**Comment by joepavitt** [comment]
@knolleary done

File: frontend/src/pages/admin/Stacks/dialogs/AdminStackEditDialog.vue [hunk 1]
```diff
@@ -144,59 +144,62 @@ export default {
...
-                opts.properties[prop.name] = this.input.properties[prop.name]
-            })
-
-            if (this.stack) {
-                if (this.editDisabled) {
-                    opts = { active: this.input.active }
-                    if (!this.editTypeDisabled && this.input.projectType) {
-                        opts.projectType = this.input.projectType
-                    }
+            console.log('confirm')
```
Comment (knolleary) [inline frontend/src/pages/admin/Stacks/dialogs/AdminStackEditDialog.vue:147]:
```suggestion
```

File: frontend/src/pages/project/Settings/dialogs/ConfirmProjectDeleteDialog.vue [hunk 2]
```diff
@@ -50,15 +50,18 @@ export default {
...
+                this.isOpen = false
+            }
         }
     },
     setup () {
         const isOpen = ref(false)
         return {
             isOpen,
             close () {
+                console.log('close')
```
Comment (knolleary) [inline frontend/src/pages/project/Settings/dialogs/ConfirmProjectDeleteDialog.vue:64]:
```suggestion
```

**Review by knolleary — APPROVED** [review]
Couple console.log's to squash, but otherwise LGTM

**Pull request merged**